packages/material-ui-system/src/index.spec.tsx

```diff
@@ -22,6 +23,39 @@ function composeTest() {
   styler({ color: 'test', spacing: 1 });
 }
 
+function styleOptionalPropsTest() {
+  const customStyle = style({ prop: 'fontColor', cssProperty: 'color' });
+  const StyledComponent = muiStyled('div')(customStyle);
+
+  // prop 'fontColor' must not be required
+  <StyledComponent />;
+  // prop 'theme' must not be required - it can be passed via context, or by styled enhancer via defaultTheme
```

Comment:
I don't think it can be passed via context. It's read from `props` only:

https://github.com/eps1lon/material-ui/blob/86012a26d49c1d9bcb480ab50da3f6add82228b7/packages/material-ui-system/src/style.js#L21

Reply:
The style function you have linked has indeed nothing to do with react/components and therefore can not get context. Its simply a function which expects plain js object as argument.

However the test does not call style directly but uses a component enhanced with `styled` using `style`-created `StyleFunction`. The component can in fact get the theme via context:

https://codesandbox.io/s/stoic-yonath-3ukmk

Anyway, you are actually still right. This is again the `styled` or `style` question. If not the built-ins I would have gone straight for the `styled` but since the question was unanswered I picked the `conform to built-ins` approach to start with.